Client registration

beganovich opened the pull request:
This will enable clients registration on their own. 

Notes:

**Subdomain or key:**
For hosted & configured self-hosted with multi-domains we'll use `amazing-company.invoiceninja.com/client/register` so URL remains clean. For others `company_key` must be appended. 

Example: `http://my-ninja.com/client/register/company-key-here`

**Workflow:**
Domains have priority. If they're not presented and no key, the page will be 404.

beganovich commented:
Preview:

![image](https://user-images.githubusercontent.com/13711415/83164147-2cb59a80-a10c-11ea-82af-f025655d64bd.png)

turbo124 commented:
This looks great! I think we should include all the client fields so that it can be a complete record.

hillelcoren commented:
Really nice!!

beganovich commented:
Changes:
- Added company logo at the top of the page
- Added support for a popup with the custom privacy policy & terms of service

![image](https://user-images.githubusercontent.com/13711415/85293766-1a761480-b49e-11ea-9905-79b2d63d1f95.png)

hillelcoren commented:
This is so awesome!!! 

@turbo124 we should probably make this a configurable setting

turbo124 commented:
@hillelcoren 

for sure, variables in settings

```php
    public $client_can_register = false;
    public $client_signup_terms = '';
    public $client_signup_privacy_policy = '';
```

turbo124 commented:
@beganovich nice work, looks great!
